tests/unit/test_utils.py

```diff
@@ -82,3 +82,31 @@ def test_invalid_port_mapping(port_spec):
         utils.validate_and_generate_port_mapping([port_spec])
 
     assert 'Port specification "{}"'.format(port_spec) in str(e.value)
+
+
+def test_deep_get():
+    data = {"data": {"files": [1, 2, 3]}}
+    assert utils.deep_get(data, "data.files.0") == 1
+    assert utils.deep_get(data, "data.files.1") == 2
+    assert utils.deep_get(data, "data.files") == [1, 2, 3]
+    assert utils.deep_get(data, "data") == {"files": [1, 2, 3]}
+
+
+def test_is_doi():
+    assert utils.is_doi("10.1234/jshd123") != None
```

Comment:
For even more Pythonicness people usually do `something is None` and `something is not None` instead of `==` and `!=`. To be honest I'd have to dig a bit to remember/explain why or in which edge cases `is` is better :D